File: frontend/store/model/shared.ts

export class WithFromJson {

  constructor() {
  }

  fromJson(data: object) {
    Object.assign(this, data);
    return this;
  }

  isObjectWithKey(data: object, key: string) {
    try {
      let obj = {}
      Object.assign(obj, data);
      if (!(key in obj)) {
        return false
      }
    } catch (e) {
      return false;
    }
    return true;
  }
}

export class Status extends WithFromJson {
  public success: boolean;
  public error_code: string;
  public error_message: string;

  constructor(success: boolean = false, error_code: string = "", error_message: string = "") {
    super();
    this.success = success;
    this.error_code = error_code;
    this.error_message = error_message
  }
}

export class FormValidation extends WithFromJson {
  public errors: {[key: string]: Array<string>};
  public passed: boolean;


  constructor(errors: {[key: string]: Array<string>} = {}, passed: boolean = false) {
    super();
    this.errors = errors;
    this.passed = passed;
  }
}

export class Auth extends WithFromJson {
  public jwt: string;
  public refresh_token: string;

  constructor(jwt: string = "", refresh_token: string = "") {
    super();
    this.jwt = jwt;
    this.refresh_token = refresh_token;
  }
}

File: web-api/chalicelib/model/shared.py
from dataclasses import dataclass
from dataclasses_json import dataclass_json
from chalicelib.model.constants import ErrorCode
from chalicelib.i18n import te


@dataclass_json
@dataclass
class Status:
    """Status of processing of the request
    """
    success: bool = False
    error_code: ErrorCode = ErrorCode.NONE
    error_message: str = ""

    def error(self, error_code: ErrorCode, error_message: str):
        self.error_code = error_code
        self.error_message = error_message

    def generic_error(self):
        self.error_code = ErrorCode.GENERIC_ERROR
        self.error_message = te("generic.error")


@dataclass_json
@dataclass
class Auth:
    """Auth data for the request
    """
    jwt: str = ""
    refresh_token: str = ""


@dataclass_json
@dataclass
class OnlyStatus:
    """Responses that only return status
    """
    status: Status = Status()
